cmd/promql-langserver/promql-langserver.go

```diff
@@ -37,12 +52,27 @@ func main() {
 	}
 	if config.RESTAPIPort != 0 {
 		fmt.Fprintln(os.Stderr, "REST API: Listening on port ", config.RESTAPIPort)
+
+		r := prometheus.NewRegistry()
+		r.MustRegister(httpRequestsTotal)
+		r.MustRegister(httpRequestDuration)
+
 		handler, err := rest.CreateHandler(context.Background(), config.PrometheusURL)
 		if err != nil {
 			log.Fatal(err)
 		}
+		handlerFunc := func(name string) http.Handler {
+			return promhttp.InstrumentHandlerDuration(
+				httpRequestDuration.MustCurryWith(prometheus.Labels{"handler": name}),
+				promhttp.InstrumentHandlerCounter(httpRequestsTotal, handler))
+		}
+		http.Handle("/diagnostics", handlerFunc("diagnostics"))
+		http.Handle("/signatureHelp", handlerFunc("signatureHelp"))
+		http.Handle("/completion", handlerFunc("completion"))
+		http.Handle("/hover", handlerFunc("hover"))
+		http.Handle("/metrics", promhttp.HandlerFor(r, promhttp.HandlerOpts{}))
 
-		err = http.ListenAndServe(fmt.Sprint(":", config.RESTAPIPort), handler)
+		err = http.ListenAndServe(fmt.Sprint(":", config.RESTAPIPort), nil)
```

Comment:
Let's avoid using the default http handler

Reply:
Does using a separate handler for the metrics endpoint make sense then?

Reply:
Yes.